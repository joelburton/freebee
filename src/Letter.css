.Letter {
  height: 100px;
  width: 100px;
  font-size: 35px;
  position: absolute;
  border-radius: 50%;
  text-align: center;
  background-color: silver;
}

.Letter-isCenter {
  background-color: gold;
}

.Letter span {
  top: 26px;
  position: relative;
  text-transform: uppercase;
}
