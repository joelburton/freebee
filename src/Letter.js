import "./Letter.css";

/** Show a single letter.
 *
 * Props:
 * - letter
 * - isCenter: true if is the center letter
 *
 * Letters -> Letter
 *
 */

function Letter({ letter, isCenter }) {
  console.debug("* Letter", letter, isCenter);
  return (
    <div className={`Letter ${isCenter ? "Letter-isCenter" : ""}`}>
      <span>{letter}</span>
    </div>
  );
}

export default Letter;
