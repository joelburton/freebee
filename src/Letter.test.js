import { render } from "@testing-library/react";
import Letter from "./Letter";

it("renders without crashing", function () {
  render(<Letter letter="A" />);
});

it("shows letter", function () {
  const { container } = render(<Letter letter="A" />);
  const letter = container.querySelector(".Letter");
  expect(Object.values(letter.classList)).toEqual(["Letter"]);
});

it("shows center letter", function () {
  const { container } = render(<Letter letter="A" isCenter />);
  const letter = container.querySelector(".Letter");
  expect(Object.values(letter.classList)).toEqual([
    "Letter",
    "Letter-isCenter",
  ]);
});

it("matches snapshot", function () {
  const { container } = render(<Letter letter="A" />);
  expect(container).toMatchSnapshot();
});

it("matches snapshot: center letter", function () {
  const { container } = render(<Letter letter="A" isCenter />);
  expect(container).toMatchSnapshot();
});
