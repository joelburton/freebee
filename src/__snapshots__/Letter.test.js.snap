// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="Letter "
  >
    <span>
      A
    </span>
  </div>
</div>
`;

exports[`matches snapshot: center letter 1`] = `
<div>
  <div
    class="Letter Letter-isCenter"
  >
    <span>
      A
    </span>
  </div>
</div>
`;
